Consolidate the use of URL parameters to specify database override

jianglai opened the pull request:
There are actions for which we want to provide an override for the database
to use, like when launching Spec11 and Invoicing pipelines. It make sense to
consolidate around the same parameter provided from the same module for
consistency in all cases, instead of defining an override for each action.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/google/nomulus/1331)
<!-- Reviewable:end -->

gbrodman reviewed (approved):
> Reviewed 10 of 13 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1331)* status: all files reviewed, 3 unresolved discussions (waiting on @jianglai and @sarahcaseybot)

___
*[core/src/main/java/google/registry/reporting/billing/GenerateInvoicesAction.java, line 95 at r2](https://reviewable.io/reviews/google/nomulus/1331#-MkCxusbCTwI_YZHziXH:-MkCxusc7qT2XSSLh4eg:b-d9ebsl) ([raw file](https://github.com/google/nomulus/blob/38010705825477e35428bc3709a23d8d10973a5c/core/src/main/java/google/registry/reporting/billing/GenerateInvoicesAction.java#L95)):*
> ```Java
>     this.jobRegion = jobRegion;
>     this.stagingBucketUrl = stagingBucketUrl;
>     this.database = database;
> ```

super nitty but we probably want to keep this in the same order as the parameters, so this would be after shouldPublish
___
*[core/src/main/java/google/registry/reporting/spec11/GenerateSpec11ReportAction.java, line 91 at r2](https://reviewable.io/reviews/google/nomulus/1331#-MkCy9P_3r-y3_PiWIG2:-MkCy9P_3r-y3_PiWIG3:btb16xc) ([raw file](https://github.com/google/nomulus/blob/38010705825477e35428bc3709a23d8d10973a5c/core/src/main/java/google/registry/reporting/spec11/GenerateSpec11ReportAction.java#L91)):*
> ```Java
>     this.jobRegion = jobRegion;
>     this.stagingBucketUrl = stagingBucketUrl;
>     this.database = database;
> ```

same nit as above, probably don't need to move this line
___
*[core/src/test/java/google/registry/reporting/billing/GenerateInvoicesActionTest.java, line 69 at r2](https://reviewable.io/reviews/google/nomulus/1331#-MkCzTi3CCFETIjW2dbN:-MkCzTi3CCFETIjW2dbO:bu8x0fy) ([raw file](https://github.com/google/nomulus/blob/38010705825477e35428bc3709a23d8d10973a5c/core/src/test/java/google/registry/reporting/billing/GenerateInvoicesActionTest.java#L69)):*
> ```Java
>     assertThat(response.getContentType()).isEqualTo(MediaType.PLAIN_TEXT_UTF_8);
>     assertThat(response.getStatus()).isEqualTo(SC_OK);
>     assertThat(response.getPayload()).isEqualTo("Launched dataflow template.");
> ```

I think these need changing for the test to pass


<!-- Sent from Reviewable.io -->

jianglai reviewed (commented):
> Reviewed 11 of 13 files at r1, 3 of 3 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1331)* status: all files reviewed, 3 unresolved discussions (waiting on @gbrodman and @sarahcaseybot)

___
*[core/src/main/java/google/registry/reporting/billing/GenerateInvoicesAction.java, line 95 at r2](https://reviewable.io/reviews/google/nomulus/1331#-MkCxusbCTwI_YZHziXH:-MkD7RCdCFqZzUBMtn37:bs93tix) ([raw file](https://github.com/google/nomulus/blob/38010705825477e35428bc3709a23d8d10973a5c/core/src/main/java/google/registry/reporting/billing/GenerateInvoicesAction.java#L95)):*
<details><summary><i>Previously, gbrodman wrote…</i></summary><blockquote>

super nitty but we probably want to keep this in the same order as the parameters, so this would be after shouldPublish
</blockquote></details>

You would be right, except that database is actually used below to define billingBucketUrl... We can change the constructor parameter order, but it doesn't seem worth the effort give that this parameter will go away eventually.
___
*[core/src/main/java/google/registry/reporting/spec11/GenerateSpec11ReportAction.java, line 91 at r2](https://reviewable.io/reviews/google/nomulus/1331#-MkCy9P_3r-y3_PiWIG2:-MkD7nQiEb9OgCB4KECC:bs4q9op) ([raw file](https://github.com/google/nomulus/blob/38010705825477e35428bc3709a23d8d10973a5c/core/src/main/java/google/registry/reporting/spec11/GenerateSpec11ReportAction.java#L91)):*
<details><summary><i>Previously, gbrodman wrote…</i></summary><blockquote>

same nit as above, probably don't need to move this line
</blockquote></details>

Ditto.
___
*[core/src/test/java/google/registry/reporting/billing/GenerateInvoicesActionTest.java, line 69 at r2](https://reviewable.io/reviews/google/nomulus/1331#-MkCzTi3CCFETIjW2dbN:-MkDPaxz31PNP2KNyBHD:b-376vyb) ([raw file](https://github.com/google/nomulus/blob/38010705825477e35428bc3709a23d8d10973a5c/core/src/test/java/google/registry/reporting/billing/GenerateInvoicesActionTest.java#L69)):*
<details><summary><i>Previously, gbrodman wrote…</i></summary><blockquote>

I think these need changing for the test to pass
</blockquote></details>

Done. Good catch.


<!-- Sent from Reviewable.io -->

gbrodman reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1331)* status: all files reviewed, 3 unresolved discussions (waiting on @gbrodman, @jianglai, and @sarahcaseybot)

___
*[core/src/main/java/google/registry/reporting/billing/GenerateInvoicesAction.java, line 95 at r2](https://reviewable.io/reviews/google/nomulus/1331#-MkCxusbCTwI_YZHziXH:-MkDT6HV45kybEFhT4Ie:b1k9nme) ([raw file](https://github.com/google/nomulus/blob/38010705825477e35428bc3709a23d8d10973a5c/core/src/main/java/google/registry/reporting/billing/GenerateInvoicesAction.java#L95)):*
<details><summary><i>Previously, jianglai (Lai Jiang) wrote…</i></summary><blockquote>

You would be right, except that database is actually used below to define billingBucketUrl... We can change the constructor parameter order, but it doesn't seem worth the effort give that this parameter will go away eventually.
</blockquote></details>

`database` is, but `this.database` isn't, right? 


<!-- Sent from Reviewable.io -->

jianglai commented:
> > _[Reviewable](https://reviewable.io/reviews/google/nomulus/1331)_ status: all files reviewed, 3 unresolved discussions (waiting on @gbrodman, @jianglai, and @sarahcaseybot)
> 
> _[core/src/main/java/google/registry/reporting/billing/GenerateInvoicesAction.java, line 95 at r2](https://reviewable.io/reviews/google/nomulus/1331#-MkCxusbCTwI_YZHziXH:-MkDT6HV45kybEFhT4Ie:b1k9nme) ([raw file](https://github.com/google/nomulus/blob/38010705825477e35428bc3709a23d8d10973a5c/core/src/main/java/google/registry/reporting/billing/GenerateInvoicesAction.java#L95)):_
> 
> Previously, jianglai (Lai Jiang) wrote…
> `database` is, but `this.database` isn't, right?

Lol you're right. I think I moved it before further refactoring that makes the move unnecessary. Fixed.

gbrodman reviewed (approved):
> Reviewed 2 of 4 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1331)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @sarahcaseybot)



<!-- Sent from Reviewable.io -->

jianglai reviewed (commented):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1331)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jianglai)



<!-- Sent from Reviewable.io -->
